airbyte-integrations/connectors/destination-meilisearch/src/main/java/io/airbyte/integrations/destination/meilisearch/MeiliSearchDestination.java

```diff
@@ -0,0 +1,190 @@
+import io.airbyte.integrations.DefaultSpecConnector;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.DestinationConsumer;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.destination.buffered_stream_consumer.BufferedStreamConsumer;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.CatalogHelpers;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.SyncMode;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * This is our first destination that is not a relational database. It therefore makes some slightly
+ * choices. The main difference that we need to reckon with is that this destination does not work
+ * without a primary key for each stream. That primary key needs to be defined ahead of time. Only
+ * records for which that primary key is present can be uploaded. There are also some rules around
+ * the allowed formats of these primary keys.
+ * </p>
+ * <p>
+ * The strategy is to inject an extra airbyte primary key field in each record. The value of that
+ * field is a randomly generate UUID. This means that we have no ability to ever overwrite
+ * individual records that we put in MeiliSearch.
+ * </p>
+ * <p>
+ * Index names can only contain alphanumeric values, so we normalize stream names to meet these
+ * constraints. This is why streamName and indexName are treated separately in this connector.
+ * </p>
+ * <p>
+ * This destination can support full refresh and incremental. It does NOT support normalization. It
+ * breaks from the paradigm of having a "raw" and "normalized" table. There is no DBT for
+ * MeiliSearch so we write the data a single time in a way that makes it most likely to work well
+ * within MeiliSearch.
+ * </p>
+ */
+public class MeiliSearchDestination extends DefaultSpecConnector implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MeiliSearchDestination.class);
+
+  public static final String AB_PK_COLUMN = "_ab_pk";
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      LOGGER.info("config in check {}", config);
+      // create a fake index and add a record to it to make sure we can connect and have write access.
+      final Client client = getClient(config);
+      final Index index = client.index("_airbyte");
+      index.addDocuments("[{\"id\": \"_airbyte\" }]");
+      index.search("_airbyte");
```

Comment:
can we also remove it to clean up side effects?

Reply:
good idea.